ParallelMatrixMultiplication/ParallelMatrixMultiplication/Matrix.cs

```diff
@@ -0,0 +1,251 @@
+    /// <exception cref="ArgumentException">Exception if matrices cannot be multiplied.</exception>
+    public static Matrix ParallelMultiplication(Matrix firstMatrix, Matrix secondMatrix)
+    {
+        if (firstMatrix.GetColumnsCount != secondMatrix.GetRowsCount)
+        {
+            throw new ArgumentException("The number of columns of the first matrix is not equal to the number of rows of the second matrix.");
+        }
+
+        var threadCount = Math.Min(Environment.ProcessorCount, firstMatrix.GetRowsCount);
+        var threads = new Thread[threadCount];
+        var chunkSize = (firstMatrix.GetRowsCount / threads.Length) + 1;
+
+        var currentMatrix = new int[firstMatrix.GetRowsCount, secondMatrix.GetColumnsCount];
+
+        for (int i = 0; i < threads.Length; i++)
+        {
+            var local = i;
+            threads[i] = new Thread(() =>
+            {
+                for (int j = local * chunkSize;
+                    j < (local + 1) * chunkSize && j < firstMatrix.GetRowsCount; j++)
+                {
+                    for (int k = 0; k < secondMatrix.GetColumnsCount; k++)
+                    {
+                        for (int p = 0; p < secondMatrix.GetRowsCount; p++)
+                        {
+                            currentMatrix[j, k] += firstMatrix[j, p] * secondMatrix[p, k];
+                        }
+                    }
+                }
+            });
+        }
+
+        foreach (var thread in threads)
+        {
+            thread.Start();
+        }
+
+        foreach (var thread in threads)
+        {
+            thread.Join();
+        }
+
+        return new Matrix(currentMatrix);
+    }
+
+    /// <summary>
+    /// A method for checking the equivalence of matrices.
+    /// </summary>
+    /// <param name="first">first matrix.</param>
+    /// <param name="second">second matrix.</param>
+    /// <returns>Are matrices equivalent.</returns>
+    public static bool Equals(Matrix first, Matrix second)
+    {
+        if (first.GetRowsCount != second.GetRowsCount ||
+            first.GetColumnsCount != second.GetColumnsCount)
+        {
+            return false;
+        }
+
+        for (int i = 0; i < first.GetRowsCount; i++)
+        {
+            for (int j = 0; j < first.GetColumnsCount; j++)
+            {
+                if (first[i, j] != second[i, j])
+                {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Matrix printing method.
+    /// </summary>
+    public void Print()
+    {
+        foreach (var line in this.matrix)
+        {
+            foreach (var num in line)
+            {
+                Console.Write($"{num} ");
+            }
+
+            Console.WriteLine();
+        }
+    }
+
+    /// <summary>
+    /// Gets rows count.
+    /// </summary>
+    public int GetRowsCount
+    {
+        get => this.matrix.Count;
+    }
```

Comment:
```suggestion
    public int RowsCount => this.matrix.Count;
```
Свойства, в отличие от методов, не именуются глаголом, поскольку не предполагают действие. Ну а `get => ` для expression-bodied-свойства без сеттера писать просто излишне, синтаксис через `=>` как раз геттер и сгенерит